include/camp/array.hpp

```diff
@@ -0,0 +1,291 @@
+      }
+
+      CAMP_HOST_DEVICE constexpr const_iterator end() const noexcept {
+         return &elements[size()];
+      }
+
+      CAMP_HOST_DEVICE constexpr const_iterator cend() const noexcept {
+         return &elements[size()];
+      }
+
+      CAMP_HOST_DEVICE constexpr bool empty() const noexcept {
+         return size() == 0;
+      }
+
+      CAMP_HOST_DEVICE constexpr size_type size() const noexcept {
+         return N;
+      }
+
+      CAMP_HOST_DEVICE constexpr size_type max_size() const noexcept {
+         return size();
+      }
+
+      CAMP_HOST_DEVICE constexpr void fill(const T& value) {
+         for (std::size_t i = 0; i < size(); ++i) {
+            elements[i] = value;
+         }
+      }
+
+      value_type elements[N];
+   };
+
+   template <class T, std::size_t N>
+   CAMP_HOST_DEVICE constexpr bool operator==(const array<T, N>& lhs,
+                                              const array<T, N>& rhs) {
+      for (std::size_t i = 0; i < N; ++i) {
+         if (lhs[i] != rhs[i]) {
+            return false;
+         }
+      }
+
+      return true;
+   }
+
+   template <class T, std::size_t N>
+   CAMP_HOST_DEVICE constexpr bool operator!=(const array<T, N>& lhs,
+                                              const array<T, N>& rhs) {
+      return !(lhs == rhs);
+   }
+
+   template <class T, std::size_t N>
+   CAMP_HOST_DEVICE constexpr bool operator<(const array<T, N>& lhs,
+                                             const array<T, N>& rhs) {
+      for (std::size_t i = 0; i < N; ++i) {
+         if (lhs[i] < rhs[i]) {
+            return true;
+         }
+         else if (lhs[i] == rhs[i]) {
+            continue;
+         }
+         else {
+            return false;
+         }
+      }
+
+      return false;
+   }
+
+   template <class T, std::size_t N>
+   CAMP_HOST_DEVICE constexpr bool operator<=(const array<T, N>& lhs,
+                                              const array<T, N>& rhs) {
+      for (std::size_t i = 0; i < N; ++i) {
+         if (lhs[i] < rhs[i]) {
+            return true;
+         }
+         else if (lhs[i] == rhs[i]) {
+            continue;
+         }
+         else {
+            return false;
+         }
+      }
+
+      return true;
+   }
+
+   template <class T, std::size_t N>
+   CAMP_HOST_DEVICE constexpr bool operator>(const array<T, N>& lhs,
+                                             const array<T, N>& rhs) {
+      for (std::size_t i = 0; i < N; ++i) {
+         if (lhs[i] > rhs[i]) {
+            return true;
+         }
+         else if (lhs[i] == rhs[i]) {
+            continue;
+         }
+         else {
+            return false;
+         }
+      }
+
+      return false;
+   }
+
+   template <class T, std::size_t N>
+   CAMP_HOST_DEVICE constexpr bool operator>=(const array<T, N>& lhs,
+                                              const array<T, N>& rhs) {
+      for (std::size_t i = 0; i < N; ++i) {
+         if (lhs[i] > rhs[i]) {
```

Comment:
```suggestion
         if (lhs[i] >= rhs[i]) {
```

Similar question as in the `<=` operator case.